In [1]:
# Import the necessary dependencies.
import numpy as np
import pandas as pd
import plotly.express as px
import scipy.stats as sts
import plotly.express as plt
import statsmodels.api as sm

## I will perform analysis of two questions:


Question 1: Does a high temperature affect the number of bike trips across the Brooklyn Bridge?

Question 2: Does the amount of precipitation affect the number of bike trips across the Brooklyn Bridge?

In [2]:
bikecounts_df = pd.read_csv("../Resources/bikecounts.csv")
bikecounts_df['Day'] = pd.to_datetime(bikecounts_df['Day'])
bikecounts_df.head()

Date        Day  High Temp (°F)  Low Temp (°F) Precipitation  \
0  4/1/16 2016-04-01            78.1           66.0          0.01   
1  4/2/16 2016-04-02            55.0           48.9          0.15   
2  4/3/16 2016-04-03            39.9           34.0          0.09   
3  4/4/16 2016-04-04            44.1           33.1      0.47 (S)   
4  4/5/16 2016-04-05            42.1           26.1             0   

   Brooklyn Bridge  Manhattan Bridge  Williamsburg Bridge  Queensboro Bridge  \
0             1704              3126                 4115               2552   
1              827              1646                 2565               1884   
2              526              1232                 1695               1306   
3              521              1067                 1440               1307   
4             1416              2617                 3081               2357   

   Total  
0  11497  
1   6922  
2   4759  
3   4335  
4   9471

## Question 1: Does a high temperature affect the number of bike trips across the Brooklyn Bridge?

Hypothesis: Higher Temp has a relationship with how many people cross Brooklyn Bridge per day

Null Hypothesis: High Temp has no association with how many people cross Brooklyn Bridge on bike

Alternative Hypothesis: High Temp correlates with how many people cross Brooklyn Bridge on bike

In [3]:
bikecounts_df.corr()
X = bikecounts_df[['High Temp (°F)']]
y = bikecounts_df['Brooklyn Bridge']
X = sm.add_constant(X)
model = sm.OLS(y, X)
results = model.fit()
print(results.summary())

                            OLS Regression Results                            
Dep. Variable:        Brooklyn Bridge   R-squared:                       0.547
Model:                            OLS   Adj. R-squared:                  0.530
Method:                 Least Squares   F-statistic:                     33.77
Date:                Tue, 05 Apr 2022   Prob (F-statistic):           3.05e-06
Time:                        18:09:16   Log-Likelihood:                -237.29
No. Observations:                  30   AIC:                             478.6
Df Residuals:                      28   BIC:                             481.4
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                     coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------
const          -1660.4539    687.707     -2.

## Analysis:
Based on the r-squared value of 0.547, we can conclude that there is a positive relationship between cyclict crossing the Brooklyn Bridge and temperature.
More cyclist cross the Brooklyn Bridge when the temperature is higher. 
A p-value of 0.023 is less than 0.05, is also statistical evidence of the relationship. 

## Question 2: Does the amount of precipitation affect the number of bike trips across the Brooklyn Bridge?

Hypothesis: Lower Precipitation has a relationship with how many people cross Brooklyn Bridge per day

Null Hypothesis: Precipitation has no association with how many people cross Brooklyn Bridge on bike

Alternative Hypothesis: Precipitation correlates with how many people cross Brooklyn Bridge on bike

In [4]:
#Dropping all other characters except numbers to change datatype from object to int
import re
pattern = '(\S{0,4})'
bikecounts_df['Precipitation'] = bikecounts_df['Precipitation'].str.extract(pattern, flags=re.I, expand=True)
T_drop = bikecounts_df[bikecounts_df['Precipitation'] == 'T'].index
bikecounts_df = bikecounts_df.drop(T_drop, axis=0)
bikecounts_df = bikecounts_df.reset_index()

In [5]:
# Checking that only numbers exist in data 
bikecounts_df['Precipitation'].value_counts()


0       17
0.09     3
0.01     3
0.15     1
0.24     1
0.47     1
0.2      1
0.05     1
0.16     1
Name: Precipitation, dtype: int64

In [6]:
#Converting datatype to interger
bikecounts_df['Precipitation'] = pd.to_numeric(bikecounts_df['Precipitation'])

In [7]:
#Running Regression Model
bikecounts_df.corr()
X = bikecounts_df[['Precipitation']]
y = bikecounts_df['Brooklyn Bridge']
X = sm.add_constant(X)
model = sm.OLS(y, X)
results = model.fit()
print(results.summary())

                            OLS Regression Results                            
Dep. Variable:        Brooklyn Bridge   R-squared:                       0.354
Model:                            OLS   Adj. R-squared:                  0.330
Method:                 Least Squares   F-statistic:                     14.78
Date:                Tue, 05 Apr 2022   Prob (F-statistic):           0.000667
Time:                        18:09:17   Log-Likelihood:                -234.75
No. Observations:                  29   AIC:                             473.5
Df Residuals:                      27   BIC:                             476.2
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                    coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------
const          2553.2525    172.385     14.811

## Analysis:
Based on the r-squared value of 0.354, we can conclude that there is a positive relationship between cyclict crossing the Brooklyn Bridge and precipitation.
More cyclist cross the Brooklyn Bridge when the precipitation is higher. 
A p-value of 0.001 is less than 0.05, is also statistical evidence of the relationship. 

### Summary
---Precipitation and Tempreature have a positive relationship between cyclist crossing the Brooklyn Bridge
